### PR TITLE
migrate firewalld to ansible.posix

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

##### SUMMARY

migrate firewalld to ansible.posix

moved to https://github.com/ansible-collections/ansible.posix/pull/68

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

firewalld

### CODE REVIEW

File: meta/runtime.yml [hunk 1]
```diff
@@ -332,6 +332,11 @@ plugin_routing:
       deprecation:
         removal_version: 3.0.0
         warning_text: see plugin documentation for details
+    firewalld:
+      deprecation:
+        removal_version: 0.2.1
```
Comment (felixfontein) [inline meta/runtime.yml:337]:
```suggestion
        removal_version: 2.0.0
```
(This is when we will remove the redirect.)

**Comment by maxamillion** [comment]
recheck

**Comment by felixfontein** [comment]
This still needs a changelog fragment.

@maxamillion when will ansible.posix be released with the firewalld module in it?

**Comment by maxamillion** [comment]
@felixfontein +1 definitely need a changelog fragment, thank you! I can release the `ansible.posix` minor release whenever. It could even be the same day the PR is merged to pull the content into that collection if needed.

**Comment by felixfontein** [comment]
@maxamillion I'm not sure how we want to proceed with moving content after ansible_builtin_runtime.yml is frozen. I think we'll have to discuss this at the community meeting first (maybe even next week, since gundalow is away this week).

**Comment by maxamillion** [comment]
@felixfontein fair point, worst case scenario I'll maintain both places for the overlap timeframe. I'd prefer not to but if necessary, that's always an option.

**Comment by maxamillion** [comment]
For posterity: because the RC1 is delayed it was approved to do the migration now

File: meta/runtime.yml [hunk 2]
```diff
@@ -541,6 +541,11 @@ plugin_routing:
       deprecation:
         removal_version: 3.0.0
         warning_text: see plugin documentation for details
+    firewalld:
+      deprecation:
+        removal_version: 2.0.0
+        warning_text: migrated to the ansible.posix collection
+      redirect: ansible.posix.firewalld
```
Comment (felixfontein) [inline meta/runtime.yml:548]: Sorry for not realizing this earlier: you need a similar entry for the `firewalld` module_utils (in the `plugin_routing.module_utils` dict).

File: meta/runtime.yml [hunk 3]
```diff
@@ -541,6 +541,11 @@ plugin_routing:
       deprecation:
         removal_version: 3.0.0
         warning_text: see plugin documentation for details
+    firewalld:
+      deprecation:
+        removal_version: 2.0.0
+        warning_text: migrated to the ansible.posix collection
```
Comment (felixfontein) [inline meta/runtime.yml:547]:
```suggestion
        warning_text: The firewalld module has been moved to the ansible.posix collection.
```

File: changelogs/fragments/firewalld_migration.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - firewalld - migration firewalld module to ansible.posix collection
```
Comment (felixfontein) [inline changelogs/fragments/firewalld_migration.yml:3]:
```suggestion
  - firewalld - the module has been moved to the ``ansible.posix`` collection. A redirection is active, which will be removed in version 2.0.0 (https://github.com/ansible-collections/community.general/pull/623).
```

File: meta/runtime.yml (hunk 2 above)
Comment (maxamillion) [reply meta/runtime.yml:548]: no worries, thanks for catching that ... I 100% did not even notice I missed that :+1:

File: meta/runtime.yml [hunk 5]
```diff
@@ -713,6 +718,11 @@ plugin_routing:
       deprecation:
         removal_version: 2.0.0
         warning_text: The infinibox module_utils plugin has been moved to the infinidat.infinibox collection.
+    firewalld:
+      deprecation:
+        removal_version: 2.0.0
+        warning_text: The firewalld module has been moved to the ansible.posix collection.
```
Comment (felixfontein) [inline meta/runtime.yml:724]:
```suggestion
        warning_text: The firewalld module_utils has been moved to the ansible.posix collection.
```

**Comment by felixfontein** [comment]
Failures are unrelated.

**Pull request merged**